learning/tour-of-beam/backend/internal/fs_content/load.go

```diff
@@ -0,0 +1,210 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fs_content
+
+import (
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	tob "beam.apache.org/learning/tour-of-beam/backend/internal"
+)
+
+const (
+	contentInfoYaml = "content-info.yaml"
+	moduleInfoYaml  = "module-info.yaml"
+	groupInfoYaml   = "group-info.yaml"
+	unitInfoYaml    = "unit-info.yaml"
+
+	descriptionMd = "description.md"
+	hintMdRegexp  = "hint[0-9]*.md"
+)
+
+type learningPathInfo struct {
+	Sdk     string
+	Content []string `yaml:"content"`
+}
+
+type learningModuleInfo struct {
+	Id         string
+	Name       string
+	Complexity string
+	Content    []string `yaml:"content"`
+}
+
+type learningGroupInfo struct {
+	Name    string
+	Content []string `yaml:"content"`
+}
+
+type learningUnitInfo struct {
+	Id           string
+	Name         string
+	TaskName     string
+	SolutionName string
+}
+
+// Watch for duplicate ids. Not thread-safe!
+type IdsWatcher struct {
```

Comment:
Nit: This can probably be private (`idsWatcher`), right? It is pretty tightly coupled with loading content.